Foundation/NSArray.swift

```diff
@@ -589,22 +628,42 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
         
         return lastEqual ? result + 1 : result
     }
-    
+
+    public convenience init(contentsOf url: URL, error: ()) throws {
+        let plistDoc = try Data(contentsOf: url)
+        guard let plistArray = try PropertyListSerialization.propertyList(from: plistDoc, options: [], format: nil) as? Array<Any>
+            else {
+                throw NSError(domain: NSCocoaErrorDomain, code: CocoaError.propertyListReadCorrupt.rawValue, userInfo: [NSURLErrorKey : url])
+        }
+        self.init(array: plistArray)
+    }
+
+    @available(*, deprecated)
     public convenience init?(contentsOfFile path: String) {
-        self.init(contentsOfURL: URL(fileURLWithPath: path))
+        do {
+            try self.init(contentsOf: URL(fileURLWithPath: path), error: ())
+        } catch {
+            return nil
+        }
     }
-    
+
+    @available(*, deprecated)
     public convenience init?(contentsOfURL url: URL) {
```

Comment:
This one isn't right: https://developer.apple.com/documentation/foundation/nsarray/1410518-init

Reply:
Are you sure about this one? That url has the word deprecated in red letters inside a box 5 times in the top half of the page :)

Reply:
Just because it's deprecated doesn't mean it's OK for it to be wrong, does it?

Reply:
Ha, hadn't see the mis-spelling, yes I will fix that one s well